### DISCUSSION TITLE
Object is of type 'unknown' for custom command

### DISCUSSION THREAD

**Discussion by jeansberg** [cypress-io/cypress#22973] [Questions and Help] [answered]
After upgrading to Cypress 10, tests using custom commands are failing due to TS2571: Object is of type 'unknown'. In VS code I can see that 'subject' in the subject.get call is marked red, because it is of type 'unknown'.

Is there something wrong with the typing or should this command be rewritten somehow? Grateful for any help 🙏 

```
Cypress.Commands.add(
    "getSelect2OptionByText",
    { prevSubject: "optional" },
    (subject, text) =>
        subject
            ? subject.get(".select2-results").contains(text)
            : cy.get(".select2-results").contains(text)
);
```

```
declare global {
    namespace Cypress {
        interface Chainable {
            getSelect2OptionByText(
                text: string
            ): Chainable<JQuery<HTMLElement>>;
        }
    }
}
```

**Comment by jjhelguero** [comment, marked as answer]
Looks like you are not wrapping the previous subject passed into your custom command. It seems from your definition, the subject will be a jQuery element. Maybe that is the reason?

```js
Cypress.Commands.add(
    "getSelect2OptionByText",
    { prevSubject: "optional" },
    (subject, text) =>
        subject
            ? cy.wrap(subject).get(".select2-results").contains(text)
            : cy.get(".select2-results").contains(text)
);
```

**Reply by jeansberg** [reply]
This helped. Thank you!
